csharp/src/lesson02/solution/HelloActive.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+using OpenTracing.Tutorial.Library;
+
+namespace OpenTracing.Tutorial.Lesson02.Solution
+{
+    internal class HelloActive
+    {
+        private readonly ITracer _tracer;
+        private readonly ILogger<HelloActive> _logger;
+
+        public HelloActive(ITracer tracer, ILoggerFactory loggerFactory)
+        {
+            _tracer = tracer;
+            _logger = loggerFactory.CreateLogger<HelloActive>();
+        }
+
+        private string FormatString(string helloTo)
+        {
+            using (var scope = _tracer.BuildSpan("format-string").StartActive(true))
+            {
+                var helloString = $"Hello, {helloTo}!";
+                scope.Span.Log(new Dictionary<string, object>
+                {
+                    [LogFields.Event] = "string.Format",
+                    ["value"] = helloString
+                });
+                return helloString;
+            }
+        }
+
+        private void PrintHello(string helloString)
+        {
+            using (var scope = _tracer.BuildSpan("print-hello").StartActive(true))
+            {
+                _logger.LogInformation(helloString);
+                scope.Span.Log("WriteLine");
+            }
+        }
+
+        public void SayHello(string helloTo)
+        {
+            using (var scope = _tracer.BuildSpan("say-hello").StartActive(true))
+            {
+                scope.Span.SetTag("hello-to", helloTo);
+                var helloString = FormatString(helloTo);
+                PrintHello(helloString);
+            }
+        }
+
+        //public static void Main(string[] args)
```

Comment:
I am guessing something doesn't like two main functions in one module?  I suggest adding a comment above explaining that and instructing to uncomment the code to make it usable. Or maybe even better to rename the function like `SecondMain` and leave the comment to change it to `Main` before running.